### ISSUE TITLE
Cannot use geolocator plugin since last flutter upgrade (0.9.6)

### ISSUE THREAD

**Issue by fbommeau** [BaseflowIT/flutter-geolocator#121]
## 🐛 Bug Report
Can't compile geolocator plugin (due to invalid  GoogleApiAvailabilityPlugin compilation) since last upgrade to flutter 0.9.6, but works perfectly before (0.8.5):
```
.pub-cache/hosted/pub.dartlang.org/google_api_availability-1.0.2/android/src/main/kotlin/com/baseflow/googleapiavailability/GoogleApiAvailabilityPlugin.kt: (29, 48): Type inference failed. Expected type mismatch: inferred type is Boolean? but Boolean was expected
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':google_api_availability:compileDebugKotlin'.
> Compilation error. See log for more details
```

Thanks for any input !

### Configuration

**Version:** 1.x

**Platform:** 
- [ ] :iphone: iOS
- [ x] :robot: Android

**Comment by kunit1** [comment]
Seeing the same exact issue since going to Flutter 0.9.7-pre42 on the master channel

**Comment by mvanbeusekom** [comment]
@fbommeau and @kunit1, thank you very much for reporting this issue. I am on it!

**Comment by mvanbeusekom** [comment]
@fbommeau, @kunit1, we have just updated our `google_api_availability` plugin which indirectly caused the `geolocator` plugin to also fail. This problem should now be solved.

I do however have another problem when compiling the `Geolocator` plugin on the latest Flutter version (0.9.7-pre.46), for which I created a ticket at Flutter (see issue [#22721](https://github.com/flutter/flutter/issues/22721)).
